adam-core/src/main/scala/org/bdgenomics/adam/models/ReadBucket.scala

```diff
@@ -96,9 +96,9 @@ class ReadBucketSerializer extends Serializer[ReadBucket] {
 object ReadBucket {
   implicit def singleReadBucketToReadBucket(bucket: SingleReadBucket): ReadBucket = {
     val (pairedPrimary, unpairedPrimary) = bucket.primaryMapped.partition(_.getReadPaired)
-    val (pairedFirstPrimary, pairedSecondPrimary) = pairedPrimary.partition(_.getFirstOfPair)
+    val (pairedFirstPrimary, pairedSecondPrimary) = pairedPrimary.partition(_.getReadNum == 0)
```

Comment:
That seems reasonable. One of the advantages of the generic `getReadNum` is that you can handle technologies that generate more than two reads per fragment (e.g., CG), but I don't think that our deduper makes sense for non-Illumina data.